api/v1alpha1/managedfleetnotificationrecord_types.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// ManagedFleetNotificationRecordStatus defines the observed state of ManagedFleetNotificationRecord
+type ManagedFleetNotificationRecordStatus struct {
+	// Managed Fleet Notification name
+	ManagementCluster string `json:"managementCluster"`
```

Comment:
Am wondering if we need to mention the MC cluster in .status? As in, we expect to have this info in `.metadata.name` anyway correct?

Reply:
It is yet to be decided IMO, the resource name with the mc info + prefix/postfix makes more sense to me.
And I think put the mc info to the .status should be harmless.

Reply:
Ack.. Name with MC info and optional prefix/postfix can help.. Yes, adding in status is harmless.. But didn't seem to add any value IMO since that field's status won't change and will be static.. So could even be part of .spec or a label or the name of the resource itself..